### DISCUSSION TITLE
About webui translation

### DISCUSSION THREAD

**Discussion by ElMoribond** [blakeblackshear/frigate#947] [Show and tell]
I like the web interface a lot. She deserves an internationalized version.
I currently have a working version that I would like to offer.

I would like to integrate several languages into it.
At the moment I only have French.

The texts entered by the developer remain in the code.
There is only one file per additional language.

The content of the French file is below. 
If you want to modify it and / or propose another language, please drop your file here.
I would take care of integrating it as soon as possible. 

{
  "Cameras": "Caméras",
  "Events": "Évènements",
  "Debug": "Déboguer",
  "Documentation": "Documentation",
  "Autodarkmode": "Thème",
  "Light": "Clair",
  "Dark": "Sombre",
  "ToggledetectON": "Activer détection",
  "ToggledetectOFF": "Désactiver détection",
  "ToggleclipsON": "Activer clip",
  "ToggleclipsOFF": "Désactiver clip",
  "TogglesnapshotsON": "Activer cliché",
  "TogglesnapshotsOFF": "Désactiver cliché",
  "HideOptions": "Cacher options",
  "ShowOptions": "Afficher options",
  "Trackedobjects": "Objets suivis",
  "Options": "Options",
  "Regions": "Régions",
  "Motionboxes": "Boîtes de mouvement",
  "Timestamp": "Horodatage",
  "Boundingbox": "Boîte englobante",
  "MaskZonecreator": "Création de zones & masques",
  "CameraMaskZonecreator": "Création de zones & masques pour {{camera_name}}",
  "Snaptoedges": "Raccrocher aux bords",
  "Motionmasks": "Masques de mouvements",
  "Zones": "Zones",
  "Masks": "Masques",
  "Objectmasks": "Masques d'objects",
  "Displayingat": "Afficher à {{fps}} img/s",
  "Moreoptions": "Plus d'options",
  "Warning": "Avertissement",
  "WarningText": "Cet outil peut vous aider à créer des masques et des zones pour votre caméra {{camera_name}}. Une fois terminé, copiez chaque masque dans votre fichier <code className=\"font-mono\">config.yml</code> puis redémarrez votre instance Frigate pour valider vos modifications",
  "Editor": "Éditeur",
  "Camera": "Caméra",
  "Label": "Label",
  "Score": "Score",
  "Date": "Date",
  "Start": "Début",
  "End": "Fin",
  "Nomoreevents": "Aucun évènement",
  "camera": "caméra",
  "zone": "zone",
  "label": "label",
  "all": {
    "m": "tous",
    "f": "toutes"
  },
  "Timeframe": "Plage de temps ",
  "Value": "Valeur",
  "Key": "Clé",
  "Clipforevent": "Clip pour l'évènement {{event_id}}",
  "Bestimage": "Meilleure image",
  "Thumbnail": "Vignette",
  "Noclipavailable": "Aucun clip disponible ",
  "Clip": "Clip",
  "detector": "détecteur",
  "Config": "Configuration",
  "CopytoClipboard": "Copier dans le presse-papier ",
  "DebugstatsText": "Les statistiques de débogage sont mises à jour automatiquement toutes les {{secs_nb}} secondes",
  "camera_fps": "caméra img/s",
  "process_fps": "processus img/s",
  "skipped_fps": "ignoré img/s",
  "detection_fps": "détection img/s",
  "pid": "pid",
  "capture_pid": "pid de capture",
  "inference_speed": "vitesse d'inférence",
  "detection_start": "début de détection",
  "Copy": "Copier",
  "Add": "Ajouter",
  "Remove": "Supprimer",
  "Clicktoedit": "Cliquez pour modifier",
  "Addto": "Ajouter à {{obj_name}}",
  "Loading": "Chargement…"
}

**Comment by paularmstrong** [comment]
I haven't had a ton of time to think about translations yet. Doing translations "properly" will require a bit more forethought than dropping in some JSON maps, especially with likely plans to [use JSONSchema](https://github.com/blakeblackshear/frigate/pull/854) for configurations – and using descriptions from the schema to create forms and content.

**Reply by ElMoribond** [reply]
You're right, I'll be patient.
It is certainly a very good choice and it will cause a lot of change in the code.
I can't wait to see the schematic you will develop. 

Thanks for taking the time to respond. 

There were already a lot of changes I hadn't seen.
Quickly a v9 please

Congratulations to both of you for this fabulous project :)

**Reply by ElMoribond** [reply]
After reflection I do not see the relationship between translation and JSONschema.
If I understood the principle correctly, we should include all texts of all languages in the JSONschema (?).
This may increase the file size considerably.
Wouldn't it be better to keep these texts in separate files?

In addition, not all texts are linked to a configuration parameter. For example the label of the links.

I specify that I speak only of translating the WebUI.
I think translating all of the messages of Frigate core is unnecessary. Because a lot of work for little interest in my opinion.

---

I looked at the JSONschema you developed;
I believe that some parameters (I am thinking mainly of zones) could have an additional attribute.
The "enable" attribute.
This allows, later, to activate/deactivate a particular zone via the interface or MQTT.

**Reply by paularmstrong** [reply]
JSONSchema spec does not include an affordance for multiple languages.

Typically software translation systems, like [FBT](https://facebook.github.io/fbt/) are a two-step (or more) process: the first step to pull out the default-language strings (typically english) into a file that is then able to be translated to other languages. Having any strings inside JSON breaks that ability because there are no function calls available there. The second is the runtime 

Potentially the JSONSchema can be written in Python directly, then translated via function calls in there when it's requested through the API, but that means we need two different i18n frameworks – one for python and one for JS/web. This will be considerably more setup and maintenance.

What we probably need to look into is to write an actual web UI node.js server and update the nginx rules to point to it for all non-API requests. Build JSONSchema in a node.js file that's able to fun the translation function calls. At docker-image build time, we would need to build a flat .json file for the Python app to consume for validating the configuration at startup.

**Reply by Shurov** [reply]
Hi, are there any news/changes as of 2024 and v0.14?

**Reply by vsisl** [reply]
I am also interested in making frigate web UI available in more languages. I would be happy to contribute with translation to Czech language.

**Reply by theahadev** [reply]
Same here, i would like to help translating to Turkish.

**Reply by hassos246** [reply]
I could translate to Swedish.

**Reply by hawkeye217** [reply]
Translation contributions can now be made via Weblate: https://hosted.weblate.org/projects/frigate-nvr/

**Reply by hassos246** [reply]
How do I add a language?

**Reply by hawkeye217** [reply]
You'll need to sign up for a Weblate account. See the Weblate documentation: https://docs.weblate.org/en/latest/user/translating.html
